src/java/org/apache/cassandra/index/sai/disk/v1/bbtree/IntersectingPointValues.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk.v1.bbtree;
+
+import java.io.IOException;
+
+import org.apache.lucene.codecs.MutablePointValues;
+import org.apache.lucene.index.PointValues;
+import org.apache.lucene.util.BytesRef;
+
+public abstract class IntersectingPointValues extends MutablePointValues
+{
+    private static final byte[] EMPTY = new byte[0];
+
+    public boolean needsSorting()
+    {
+        return size() > 1;
+    }
+
+    public abstract void intersect(IntersectVisitor visitor) throws IOException;
+
+    @Override
+    public int getDocCount()
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public long size()
+    {
+        return 1;
+    }
```

Comment:
I'd consider a small set of changes around sorting here:

1.) Remove the `size()` method in the super right here. It's already implemented in the RAM buffer version.

2.) Have `ImmutableIntersectingPointValues` add the following:

```
@Override
    public boolean needsSorting()
    {
        return false;
    }

    @Override
    public long size()
    {
        throw new UnsupportedOperationException();
    }
```

3.) Fix `ImmutableIntersectingPointValuesTest#shouldSkipLuceneSorting()` to have a `needsSorting()` check or to hard code a size of 1.

We don't have any intention of ever sorting the immutable version, so might as well make that explicit (and avoid providing a dummy size that isn't actually a size).